src/Access/SettingsProfilesInfo.cpp

```diff
@@ -55,4 +62,26 @@ SettingsProfilesInfo::getConstraintsAndProfileIDs(const std::shared_ptr<const Se
     return res;
 }
 
+Strings SettingsProfilesInfo::getProfileNames() const
+{
+    Strings result;
+    result.reserve(profiles.size());
+    for (const auto & profile_id : profiles)
+    {
+        const auto p = names_of_profiles.find(profile_id);
+        if (p != names_of_profiles.end())
+            result.push_back(p->second);
+        else
+        {
+            if (const auto name = access_control.tryReadName(profile_id))
+                // We could've updated cache here, but it is a very rare case, so don't bother.
+                result.push_back(*name);
+            else
+                throw Exception(ErrorCodes::LOGICAL_ERROR, "Unable to get profile name for {}", toString(profile_id));
+        }
+    }
```

Comment:
@Enmk, what does it fix and how? Looks like it simply replaces `std::out_of_range` with `LOGICAL_ERROR` without fixing the root casue

Reply:
#54656

Reply:
@Enmk `SettingsProfilesInfo` must be internally consistent, so I think if we fix that consistency in https://github.com/ClickHouse/ClickHouse/pull/57263 then we only need here a logical error:

```
    for (const auto & profile_id : profiles)
    {
        const auto p = names_of_profiles.find(profile_id);
        if (p != names_of_profiles.end())
            result.push_back(p->second);
        else
            throw Exception(ErrorCodes::LOGICAL_ERROR, "Unable to get profile name for {}", toString(profile_id));
    }
```